Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs

```diff
@@ -0,0 +1,141 @@
+            => GetAsync(id, new PartitionKey(partitionKeyValue ?? id), cancellationToken);
+
+        /// <inheritdoc/>
+        public async ValueTask<TItem> GetAsync(string id, PartitionKey partitionKey, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+
+            if (partitionKey == default)
+            {
+                partitionKey = new PartitionKey(id);
+            }
+
+            TItem item = Items.FirstOrDefault(i => i.Id == id && new PartitionKey(i.PartitionKey) == partitionKey);
+
+            if (item is null)
+                NotFound();
+
+            return item is { Type: { Length: 0 } } || item.Type == typeof(TItem).Name ? item : default;
+        }
+
+        /// <inheritdoc/>
+        public async ValueTask<IEnumerable<TItem>> GetAsync(Expression<Func<TItem, bool>> predicate, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+            return Items.Where(predicate.Compose(
+                item =>  item.Type == typeof(TItem).Name, Expression.AndAlso).Compile());
+        }
+
+        /// <inheritdoc/>
+        public ValueTask<IEnumerable<TItem>> GetByQueryAsync(string query, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc/>
+        public ValueTask<IEnumerable<TItem>> GetByQueryAsync(QueryDefinition queryDefinition, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc/>
+        public async ValueTask<TItem> CreateAsync(TItem value, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+
+            TItem item = Items.FirstOrDefault(i => i.Id == value.Id && i.PartitionKey == value.PartitionKey);
+
+            if(item is not null)
+                Conflict();
+
+            Items.Add(value);
+
+            return value;
+        }
+
+        /// <inheritdoc/>
+        public async ValueTask<IEnumerable<TItem>> CreateAsync(IEnumerable<TItem> values, CancellationToken cancellationToken = default)
+        {
+            IEnumerable<TItem> enumerable = values.ToList();
+
+            foreach (TItem value in enumerable)
+            {
+                await CreateAsync(value, cancellationToken);
+            }
+
+            return enumerable;
+        }
+
+        /// <inheritdoc/>
+        public async ValueTask<TItem> UpdateAsync(TItem value, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+
+            TItem item = Items.FirstOrDefault(i => i.Id == value.Id && i.PartitionKey == value.PartitionKey);
+
+            if (item is not null)
+            {
+                Items.Remove(item);
+                Items.Add(value);
+                return value;
```

Comment:
Nit: you could just omit these two lines (the `Items.Add()` and early return), since the same thing would happen if you let the method continue after this if block anyway.

Reply:
Yeah that is a good point, will amend this one.